### PR TITLE
switch: add physfs

### PR DESCRIPTION
**Pull request by Stary2001**
Cleaned up PhysFS patch using pthreads support. Continuation of #140

### CODE REVIEW

File: switch/physfs/physfs.patch [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
++		/* nx-hbmenu should give us the full path of the application, this may
++		 *  reside in a subfolder. Higher level code will strip away the name
++		 *  and extension.
++		 */
++		return NULL;
++	}
++
++	if (!retval)
++		/* Last resort: use `/switch` directory. */
++		retval = __PHYSFS_strdup("sdmc:/switch/");
```
Comment (WinterMute) [inline switch/physfs/physfs.patch:163]: Change this to just "/switch/" - avoiding explicit device names will allow code to run from different devices in the future. Is the trailing slash needed?

File: switch/physfs/physfs.patch [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
++    return __PHYSFS_switchCalcUserDir();
++#endif
+     char *retval = NULL;
+     char *envr = getenv("HOME");
+ 
+@@ -299,7 +302,14 @@
+ int __PHYSFS_platformStat(const char *fname, PHYSFS_Stat *st, const int follow)
+ {
+     struct stat statbuf;
++#ifdef PHYSFS_PLATFORM_SWITCH
```
Comment (WinterMute) [inline switch/physfs/physfs.patch:43]: This part is no longer needed since we implemented lstat via stat

File: switch/physfs/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+arch=('any')
+url='https://icculus.org/physfs/'
+license=('zlib')
+options=(!strip libtool staticlibs)
+source=( "https://icculus.org/physfs/downloads/physfs-3.0.2.tar.bz2"
+         "physfs.patch")
+makedepends=('devkitA64' 'devkitpro-pkgbuild-helpers')
+sha256sums=(
+  '304df76206d633df5360e738b138c94e82ccf086e50ba84f456d3f8432f9f863'
+  'SKIP'
```
Comment (WinterMute) [inline switch/physfs/PKGBUILD:15]: provide sha256sum value here please.

File: switch/physfs/physfs.patch (hunk 1 above)
Comment (carstene1ns) [reply switch/physfs/physfs.patch:163]: Yes, trailing slash is added everywhere else as well.
> Platform layer is required to append a dirsep

**Pull request merged**